### DISCUSSION TITLE
Regarding payment collection

### DISCUSSION THREAD

**Discussion by pkapustin** [beckn/protocol-specifications#156] [Core Specification]
We had a discussion on payment collection by BAP vs BPP. The summary is that BAP has to collect. Here are some of the main reasons. 

### Composability
The customer can only checkout once, and the price may consist of several parts.
Examples:

- Ordering a pizza, money is split between restaurant and delivery BPP
- Booking parking and bicycle rental, money is split between two BPPs
- Purchasing goods, BAP charges extra for their own service

There are plenty of examples like this from all domains (multi-leg trips, insurance, additional services, etc.)

If we were to implement a BPP integration in a way where the payment is collected by the BPP, it would have these consequences:
- The service of such a BPP would be exclusive (cannot be combined with any other services)
- No extra fees can be collected by a BAP

This would essentially prevent many new BAPs from joining the network, as it is essential for their business to compose several services, add their own services and collect extra fees.

### Uniform interface
The flows have to be the same in the same network / domain. We cannot have different types of payment collection as this would require different flows (as notifications from payment gateway would come to different parties).

### User experience
Payment collection by BAP is essential for good and consistent user experience. For this, the following is needed:
- Same UI implemented by the BAP
- Notifications from the payment gateway need to come to the BAP

BAP and BPP have totally different roles. Here is one analogy: BAP is like a software application, BPP is like a library / service that are used in an application. One application can use many services / libraries. It is only the application that can display UI.

It should be noted though that BPPs need to be educated about this, so that it is explained clearly why it has to be the BAP that collects the payments.

**Comment by techframewirk** [comment]
I agree with all the reasons mentioned above. 

But should this be a network level decision or should this be a part of the core protocol specifications itself?

**Comment by pkapustin** [comment]
I think it should apply to all networks: BAPs and BPPs have to act according to their roles. BAPs are customer-facing platforms that make the services available to the customer and combine them. BPPs provide those services. This basic premise has to hold for the network to be able to grow and provide value for the customers.

**Reply by venkatramanm** [reply]
Makes complete sense from ux perspective.


I think we still need to also look at the. COmplete flow and see how can each transaction is associated with multiple payments to be disbursed by bap. And how quickly can it be done. ? What happens if order is cancelled. Etc.  ? Every participant needs to return money.  If we need to keep ledgers., We may need a banking participant.  

More bridges to cross but yes this decision looks right.
